lib/operdebug.g

```diff
@@ -0,0 +1,69 @@
+##############################################################################
+##  operdebug.g                                                   Frank Lübeck
+##  
+##  Non-documented  utilities to  write out  infos  about all  methods of  all
+##  operations with  location and rank, and  to write out all  filters by name
+##  and their rank. These can be useful to debug larger changes that influence
+##  the ranking of filters and methods.
+##  
+BIND_GLOBAL("WriteMethodOverview", function(fname)
+  local name, sline, f;
+  name := function(f)
+    local path, ls, s;
+    path := FilenameFunc(f);
+    if path=fail then
+      return NameFunction(f);
+    fi;
+    for s in GAPInfo.RootPaths do
+      ls := Length(s);
+      if Length(path) >= ls and path{[1..ls]}=s then
+        return path{[ls+1..Length(path)]};
+      fi;
+    od;
+    return path;
+  end;
+  sline := function(f)
+    local n;
+    n := StartlineFunc(f);
+    if n=fail then
+      return "\c";
+    fi;
+    return Concatenation(":",String(n),"\c");
+  end;
+  f := function()
+    local nam, m, s, op, i, j;
+    for op in OPERATIONS do
+      nam := NameFunction(op);
+      for i in [0..6] do
+        m := METHODS_OPERATION(op, i);
+        s := 4+i;
+        for j in [1..Length(m)/s] do
+          Print(nam,";\c",i,";",
+                name(m[j*s-2]),
+                sline(m[j*s-2]),";\c",
```

Comment:
This writes out the location of the definition of the *function*, if available. It does not yet take advantage of the location of the *method installation*, which we now also track (see e.g. the code of `ApplicableMethodTypes`)